### PR TITLE
feat(typing): finalize annotations

### PR DESCRIPTION
**Pull request by CaselIT**
Finish typing falcon.
This also fixes https://github.com/falconry/falcon/issues/2207

### CODE REVIEW

File: examples/ws_tutorial/ws_tutorial/app.py [hunk 1]
```diff
@@ -1,3 +1,5 @@
+from __future__ import annotations
```
Comment (vytas7) [inline examples/ws_tutorial/ws_tutorial/app.py:1]: I think we should remove this for now, this tutorial app is not typed at all anyway.

Comment (CaselIT) [reply examples/ws_tutorial/ws_tutorial/app.py:1]: it has some types, for example:
`async def process_request_ws(self, req: Request, ws: WebSocket):`

Comment (vytas7) [reply examples/ws_tutorial/ws_tutorial/app.py:1]: Aha, checked quickly and didn't see many.
Regardless, as a user, why would I need this? I think it is useful for our own codebase, but users can use the standard mode too if they wish so?

Comment (CaselIT) [reply examples/ws_tutorial/ws_tutorial/app.py:1]: I'll add a skip in the file then

**Review by vytas7 — DISMISSED** [review]
Looks good for the most of it, I'll add a note to the cleanup issue to revise `Headers` though. :+1:

**Review by vytas7 — APPROVED** [review]

**Pull request merged**